src/terraform.ts

```diff
@@ -0,0 +1,228 @@
+import { Utils } from 'vscode-uri';
+import { getActiveTextEditor } from './utils/vscode';
+import { clientSupportsCommand } from './utils/clientHelpers';
+
+/* eslint-disable @typescript-eslint/naming-convention */
+export interface ModuleCaller {
+  uri: string;
+}
+
+export interface ModuleCallersResponse {
+  version: number;
+  moduleCallers: ModuleCaller[];
+}
+
+export interface ModuleCall {
+  name: string;
+  source_addr: string;
+  version?: string;
+  source_type?: string;
+  docs_link?: string;
+  dependent_modules: ModuleCall[];
+}
+
+export interface ModuleCallsResponse {
+  v: number;
+  module_calls: ModuleCall[];
+}
+
+interface ModuleProvidersResponse {
+  v: number;
+  provider_requirements: {
+    [provider: string]: {
+      display_name: string;
+      version_constraint?: string;
+      docs_link?: string;
+    };
+  };
+  installed_providers: {
+    [provider: string]: string;
+  };
+}
+/* eslint-enable @typescript-eslint/naming-convention */
+
+export async function moduleCallers(
+  moduleUri: string,
+  client: LanguageClient,
+  reporter: TelemetryReporter,
+): Promise<ModuleCallersResponse> {
+  const command = 'terraform-ls.module.callers';
+
+  const response = await execWorkspaceLSCommand<ModuleCallersResponse>(command, moduleUri, client, reporter);
+
+  return response;
+}
+
+export async function moduleCalls(
+  moduleUri: string,
+  client: LanguageClient,
+  reporter: TelemetryReporter,
+): Promise<ModuleCallsResponse> {
+  const command = 'terraform-ls.module.calls';
+
+  const response = await execWorkspaceLSCommand<ModuleCallsResponse>(command, moduleUri, client, reporter);
+
+  return response;
+}
+
+export async function moduleProviders(
+  moduleUri: string,
+  client: LanguageClient,
+  reporter: TelemetryReporter,
+): Promise<ModuleProvidersResponse> {
+  const command = 'terraform-ls.module.providers';
+
+  const response = await execWorkspaceLSCommand<ModuleProvidersResponse>(command, moduleUri, client, reporter);
+
+  return response;
+}
+
+export async function initCurrentDirectoryCommand(client: LanguageClient, reporter: TelemetryReporter) {
+  try {
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+    const selected = await vscode.window.showOpenDialog({
+      canSelectFiles: false,
+      canSelectFolders: true,
+      canSelectMany: false,
+      title: 'Choose which workspace to initialize with terraform init',
+      defaultUri: workspaceFolders ? workspaceFolders[0]?.uri : undefined,
+      openLabel: 'Initialize',
+    });
+    if (selected === undefined) {
+      return;
+    }
+
+    const moduleUri = selected[0];
+    const command = `terraform-ls.terraform.init`;
+
+    return execWorkspaceLSCommand<void>(command, moduleUri.toString(), client, reporter);
+  } catch (error) {
+    if (error instanceof Error) {
+      vscode.window.showErrorMessage(error instanceof Error ? error.message : error);
```

Comment:
```suggestion
      vscode.window.showErrorMessage(error.message);
```

We checked if `error` is instance of `Error` in the if clause, so it's safe to access `.message` without the extra check here.